torch/distributed/_composable/fsdp/_fsdp_common.py

```diff
@@ -72,6 +73,13 @@ class ParamModuleInfo:
     shared_param_names: List[str] = field(default_factory=list)
 
 
+class TrainingState(Enum):
+    FORWARD = auto()
```

Comment:
non-blocking: are we not distinguishing pre/post forward because forward simplier? or post-forward just set IDLE state ?

Reply:
Yes. The reason is just that distinguishing them is not useful, so we keep it just as `FORWARD` for simplicity. (`POST_FORWARD` would be the same thing as `IDLE`.)

In backward, having a `BACKWARD_POST` historically was helpful since there were two types of backward prefetching. We may be able simplify `BACKWARD_PRE` and `BACKWARD_POST` here to just `BACKWARD`, but we can see.